### PR TITLE
Setup Trivy scanning for supported release branches

### LINKED ISSUE #4394: Setup Trivy scanning for supported release branches
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
I'd like to add Trivy scanning job to all supported release branches. Same thing has been done for CAPI (https://github.com/kubernetes-sigs/cluster-api/pull/7874) and CAPV(https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1978)

It would benefit for detecting cves earlier for release branches.
**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by wyike**
**What type of PR is this?**
/kind support

**What this PR does / why we need it**:
This PR adds Trivy scanning job to all supported release branches. Same thing has been done for CAPI (https://github.com/kubernetes-sigs/cluster-api/pull/7874) and CAPV(https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1978)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4394 

**Special notes for your reviewer**:

try run in my folked repo: https://github.com/wyike/cluster-api-provider-aws/actions/runs/5699290555/job/15448043792

It depends on https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4401 

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Setup Trivy scanning for supported release branches
```

### CODE REVIEW

File: .github/workflows/scan.yml [hunk 1]
```diff
@@ -10,14 +10,23 @@ permissions: {}
...
+        with:
+          ref: ${{ matrix.branch }}
+      - name: Calculate go version
+        id: vars
+        run: echo "go_version=$(make go-version)" >> $GITHUB_OUTPUT
       - name: Setup go
         uses: actions/setup-go@v4
         with:
-          go-version: 1.20
+          go-version: ${{ steps.vars.outputs.go_version }}
```
Comment (Skarlso) [inline .github/workflows/scan.yml:30]: Why not use go-version-file and point to the go mod file?

**Review by Skarlso — COMMENTED** [review]
One question.

File: .github/workflows/scan.yml (hunk 1 above)
Comment (Skarlso) [reply .github/workflows/scan.yml:30]:
```yaml
        go-version-file: '${{ github.workspace }}/go.mod'
```

Like this.

Comment (wyike) [reply .github/workflows/scan.yml:30]: Ah, I don't know that way. It looks simpler and has no dependency on `make go-version` support.

I just followed capi and capv pattern:
https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/scan.yml#L30
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/.github/workflows/scan.yaml#L30

**Comment by Ankitasw** [comment]
/approve

Will rely on @Skarlso's lgtm 🙂

File: .github/workflows/scan.yml (hunk 1 above)
Comment (Skarlso) [reply .github/workflows/scan.yml:30]: @wyike  So.. Are you going to change it? :D

Comment (wyike) [reply .github/workflows/scan.yml:30]: sorry for the late response. Let me try.

Comment (wyike) [reply .github/workflows/scan.yml]: It becomes quite simple.

Give a try run in my folked repo: https://github.com/wyike/cluster-api-provider-aws/actions/runs/5699290555/job/15448043792

Comment (Skarlso) [reply .github/workflows/scan.yml]: Nice!

**Comment by Skarlso** [comment]
/lgtm
/hold

Remove hold once done.

**Pull request merged**